ModularContent/Sets/Template_Data.php

```diff
@@ -15,30 +15,34 @@ public function hook() {
 	 * @return array
 	 */
 	public function register_data( $data, $post ) {
-		$templates = $this->get_templates( $post );
+		$templates = $this->get_templates( $post ) ;
 		$data[ 'templates' ] = $templates;
 		return $data;
 	}
 
 	private function get_templates( $context_post ) {
-		$posts = get_posts( [
+		$posts = apply_filters( 'panel_set_get_templates', get_posts( [
```

Comment:
Similar to your approach above, I would set a var first then filter that. JIC there is some possible data or scope leakage.